theano/compile/nanguardmode.py

```diff
@@ -277,36 +264,26 @@ def do_check_on(var, nd, f, is_input):
                 elif config.NanGuardMode.action == 'warn':
                     logger.error(msg)
 
-        def nan_check(i, node, fn):
+        def nan_check(node, thunk, storage_map, compute_map):
             """
             Runs `fn` while checking its inputs and outputs for NaNs / Infs.
 
             Parameters
             ----------
-            i :
-                Currently ignored.
-                TODO: determine why it is here or remove).
             node : theano.gof.Apply
                 The Apply node currently being executed.
-            fn : callable
+            thunk : callable
                 The thunk to execute for this Apply node.
+            storage_map : dict
+                The mapping of variables to storage cells.
+            compute_map : dict
+                The mapping to get the computed state of the nodes. (ignored)
 
             """
-            inputs = fn.inputs
-            for x, var in zip(inputs, node.inputs):
```

Comment:
Because the callback is called after the node has run and there might be some inplace stuff that would make it confusing.

Also, this only check the inputs of the graph so I figure it would be easier to check those yourself.

We can bring that back with a bit of modification to StackVM, but I didn't have the time to do that.